pkg/apis/messaging/v1alpha1/pipeline_lifecycle.go

```diff
@@ -100,25 +105,32 @@ func (ps *PipelineStatus) PropagateChannelStatuses(channels []*duckv1alpha1.Chan
 		allReady = false
 
 	}
-	for i, s := range channels {
+	for i, c := range channels {
 		ps.ChannelStatuses[i] = PipelineChannelStatus{
 			Channel: corev1.ObjectReference{
-				APIVersion: s.APIVersion,
-				Kind:       s.Kind,
-				Name:       s.Name,
+				APIVersion: c.APIVersion,
+				Kind:       c.Kind,
+				Name:       c.Name,
+				Namespace:  c.Namespace,
 			},
 		}
-		address := s.Status.AddressStatus.Address
-		// TODO: Once channealables actually display status, check it here.
-		if address == nil {
+		address := c.Status.AddressStatus.Address
+		if address != nil {
+			ps.ChannelStatuses[i].ReadyCondition = apis.Condition{Type: apis.ConditionReady, Status: corev1.ConditionTrue}
```

Comment:
This isn't quite accurate, a Channelable may be addressable but not ready. 

For now, I'm OK with a TODO.

Reply:
Yes, I had a TODO there (and dropped it somewhere along the way), and I think we need to change Channelable a little bit, because it doesn't include anything useful besides this.